src/screens/Forum/ForumCategoriesScreen.js

```diff
@@ -1,10 +1,60 @@
-import React from 'react';
-import { View, Text } from 'react-native';
+import React, { useState } from 'react';
+import {
+  ScrollView, Text, View, StyleSheet,
+} from 'react-native';
+import firestore from '@react-native-firebase/firestore';
+import PropTypes from 'prop-types';
+import ForumCategoryCard from './ForumCategoryCard';
+
+const styles = StyleSheet.create({
+  container: {
+    padding: 10,
+    backgroundColor: '#daecf5',
+  },
+});
+
+
+export default function ForumCategoriesScreen({ navigation }) {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [forumCategories, setForumCategories] = useState([]);
+
+  console.log('categories ordered');
+
+  firestore()
+    .collection('forum_categories')
+    .orderBy('title')
+    .get()
+    .then((snapshot) => {
+      const forumCategoriesData = snapshot.docs.map((doc) => ({
+        ...doc.data(),
+        id: doc.id,
+      }));
+      setForumCategories(
+        forumCategoriesData
+          .map((forum) => {
+            const navigateToSubcategory = () => navigation.navigate('ForumSubcategoryPosts', { categoryID: forum.id });
+            return (
+              <ForumCategoryCard key={forum.id} navigate={navigateToSubcategory}>
+                {forum.title}
+              </ForumCategoryCard>
+            );
+          }),
+      );
+    })
+    .catch((error) => {
+      setErrorMessage(error.message);
+    });
```

Comment:
It looks like this component is being rerendered in an infinite loop (I noticed the console.log never stops), i have no idea why but putting this firestore stuff inside of `useEffect` seems to make it stop: https://reactjs.org/docs/hooks-effect.html